app/services/fan_out_on_write_service.rb

```diff
@@ -129,14 +129,19 @@ def broadcast_to_hashtag_streams!
   end
 
   def broadcast_to_public_streams!
-    return if @status.reply? && @status.in_reply_to_account_id != @account.id && !Setting.show_replies_in_public_timelines
-
-    redis.publish('timeline:public', anonymous_payload)
-    redis.publish(@status.local? ? 'timeline:public:local' : 'timeline:public:remote', anonymous_payload)
+    broadcast_to = ->(channel) {
+      redis.publish(channel, anonymous_payload)
+      if @status.with_media?
+        redis.publish("#{channel}:media", anonymous_payload)
+      end
```

Comment:
Minor style issue but I'd suggest:
```suggestion
      redis.publish("#{channel}:media", anonymous_payload) if @status.with_media?
```